[RemoteMirrors] Fix losing the remote address on StaticMirror

augusto2112 opened the pull request:
The Offset field of a DynamicRelocation is either an offset or a remote address, but was being treated only as a remote address on getDynamicSymbol.

<!--
If this pull request is targeting a release branch, please fill out the
following form:
https://github.com/swiftlang/.github/blob/main/PULL_REQUEST_TEMPLATE/release.md?plain=1

Otherwise, replace this comment with a description of your changes and
rationale. Provide links to external references/discussions if appropriate.
If this pull request resolves any GitHub issues, link them like so:

  Resolves <link to issue>, resolves <link to another issue>.

For more information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

augusto2112 commented:
@swift-ci smoke test

etcwilde commented on test/Reflection/conformance_descriptors.swift:
```diff
@@ -5,7 +5,6 @@
 //
 // Temporarily disable on AArch64 Linux (rdar://88451721)
 // UNSUPPORTED: OS=linux-gnu && CPU=aarch64
```
Do we still need this marked as unsupported on Linux?

augusto2112 replied:
That one seems unrelated to this change.

etcwilde commented on test/Reflection/typeref_decoding.swift:
```diff
@@ -10,7 +10,6 @@
 
 // FIXME: rdar://127796117
 // UNSUPPORTED: OS=linux-gnu && CPU=aarch64
```
Do we still need this marked as unsupported on Linux?

augusto2112 replied:
Same as above.

finagolfin replied:
[Same passing claim as above](https://github.com/swiftlang/swift/pull/82325#issuecomment-3002136570), can probably be checked by enabling them and running the linux AArch64 CI on this pull.

augusto2112 commented:
@swift-ci smoke test

mikeash commented on include/swift/StaticMirror/ObjectFileContext.h:
```diff
@@ -39,7 +39,11 @@ class Image {
     StringRef Symbol;
+    // The offset (if the symbol is available), or the resolved remote address
+    // if the symbol is empty.
     uint64_t Offset;
```
Should we rename this to `OffsetOrAddress` or something like that?

drodriguez commented:
I can confirm applying this locally also fixes those two tests in my setup. Thanks!

etcwilde commented:
> I don't think that failure is related to this change

I agree. Just had a script watch the PR for failed builds and extract failures for documenting purposes.
One of the radars calls out UBI-9 explicitly, so probably worth running that job too.

etcwilde commented:
Yeah, I'm not sure we have a trigger for aarch64 UBI9. Was worth a try.

augusto2112 committed:
[RemoteMirrors] Fix losing the remote address on StaticMirror

The Offset field of a DynamicRelocation is either an offset or a remote
address, but was being treated only as a remote address on
getDynamicSymbol.

augusto2112 commented:
@swift-ci smoke test

augusto2112 commented:
I want to cherry-pick this into 6.2, so to play it safe, I'll keep those tests unsupported in this PR, and open a separate PR reenabling those tests after this is merged.

finagolfin commented:
> Yeah, I'm not sure we have a trigger for aarch64 UBI9.

There is [a UBI 9 aarch64 CI job](https://ci.swift.org/job/oss-swift-package-ubi-9-aarch64/), so it would flag the issue once committed, if this still didn't work.

Not sure about manual pre-commit CI triggers though, is there a list somewhere public? [The official doc is way behind](https://github.com/swiftlang/swift/blob/main/docs/ContinuousIntegration.md).

augusto2112 commented:
I'm enabling the tests here https://github.com/swiftlang/swift/pull/83187

finagolfin commented:
Can confirm [these tests are passing on Android](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-24.04-android-arm64/574/consoleText) again, thanks @augusto2112. 😃
